network/lite/src/client.rs

```diff
@@ -0,0 +1,210 @@
+
+  fn syn_bytes() -> Vec<u8> {
+    let mut connect_msg = LiteMessage::new();
+    connect_msg.set_packet_type(LiteMessage_Type::SYN);
+    connect_msg.write_to_bytes().expect("packet wasn't serializable")
+  }
+
+  fn ack_bytes(our_id: u32) -> Vec<u8> {
+    let mut connect_msg = LiteMessage::new();
+    connect_msg.set_packet_type(LiteMessage_Type::ACK);
+    connect_msg.set_origin_id(our_id);
+    connect_msg.write_to_bytes().expect("packet wasn't serializable")
+  }
+
+  fn data_bytes(our_id: u32, payload: [u8; protocol::PAYLOAD_MAX_SIZE]) -> Vec<u8> {
+    let mut connect_msg = LiteMessage::new();
+    connect_msg.set_origin_id(our_id);
+    connect_msg.set_packet_type(LiteMessage_Type::DATA);
+    connect_msg.set_data(payload.to_vec());
+    connect_msg.write_to_bytes().expect("packet wasn't serializable")
+  }
+
+  fn unknown_sender_bytes(&self) -> Vec<u8> {
+    let mut connect_msg = LiteMessage::new();
+    connect_msg.set_packet_type(LiteMessage_Type::UNKNOWN_SENDER);
+    connect_msg.set_origin_id(self.connection.get_id());
+    connect_msg.write_to_bytes().expect("packet wasn't serializable")
+  }
+
+  pub fn connect(&mut self) -> bool {
+    use protocol::ConnectionStatus::*;
+    use protocol::ClientNegotiationStatus::*;
+
+    self.cull_dead_connections();
+    let mut tries: u32 = 0;
+    let mut failed = false;
+    let mut deadline = SystemTime::now()
+      + Duration::from_millis(protocol::SYN_ACK_WAIT_MS as u64);
+
+    let mut now = SystemTime::now();
+    while !failed
+        && !self.connection.is_connected() {
+      while now < deadline
+          && !self.connection.is_connected() {
+        let possible_lite_msg = self.internal_socket.recv()
+          .expect("broken gaffer pipe")
+          .and_then(|pkt| if pkt.addr == self.server { Some(pkt) } else { None })
+          .and_then(|pkt| protobuf::parse_from_bytes::<LiteMessage>(&pkt.payload).ok());
+
+        match (&self.negotiation, possible_lite_msg, tries) {
+          (_, _, protocol::SYN_ACK_RETRIES) => {
+            failed = true;
+          },
+          // Send Syn
+          (&None, _, _) => {
+            self.internal_socket.send(GafferPacket::new(self.server.clone(), LiteClient::syn_bytes()));
+            self.negotiation = Some(ClientNegotiationStatus::ClientSyn);
+            tries += 1;
+          },
+          // Respond to Syn+Ack
+          (&Some(ClientSyn), Some(ref lite_msg), _)
+              if lite_msg.get_packet_type() == LiteMessage_Type::SYN_ACK
+              && lite_msg.get_set_your_id() != 0 => {
+            let our_id = lite_msg.get_set_your_id();
+            self.internal_socket.send(GafferPacket::new(self.server.clone(), LiteClient::ack_bytes(our_id)));
+            self.negotiation = None;
+            self.connection = Connected { id: our_id, last_message: now };
+          },
+          (_, None, _) => {
+            // Nothin to do, no packet
+          }
+          _ => {
+            // Some weird packets or something
```

Comment:
hmm....

Could be....

Not sure what to log though. "Got weird packet?" Thats not actionable really.